lib/src/configuration.dart

```diff
@@ -49,6 +49,11 @@ typedef InitialDataCallback = void Function(Realm realm);
 /// Client reset errors will not be reported through this callback as they are handled by [ClientResetHandler].
 typedef SessionErrorHandler = void Function(SessionError error);
 
+/// The signature of a callback that will be invoked if a client reset error occurs for this [Realm].
+///
+/// Currently, Flexible sync only supports the Manual Recovery.
```

Comment:
Even though we only support manual reset, we should design the API in a way that is extensible. My suggestion is to model it the same way as the .NET API where we have an abstract handler and concrete implementations (dart will only have one to start with, but more will be added later).

Reply:
You are right. We can also only rename the handler to ManualyClientResetHandler in order to avoid breaking changes later.